## Results

![**Max Containment to ANI and AAI.** 
Containment calculation is guaranteed to be more similar to traditional calculation of Average Nucleotide Identity and Average Amino Acid Identity, which compared only the sections of genome that align. The shared k-mer content (containment numerator) can be thought of as the alignable sections of the genomes. 
The denominator of the Jaccard index is the alignable sections + the unalignable sections. 
The lower bound of the containment denominator will be the exact same as the numerator at 100% containment, where all k-mers are found within the comparison dataset. The upper bound will be the same as the Jaccard denominator, where all k-mers of the comparison dataset are found within the query dataset, and it is the query that contains any additional nonshared k-mers/unalignable sequence.
](images/containment-ANI-AAI.png){#fig:containmentANI}


Main q's to answer:
1. We promise it works
   - how does it compare with gold standard? (ANIb/ANIm)
   - how does it compare with current alignment-free methods (Mash, FastANI)
   - how does it function on contaminated and/or incomplete genomes?
   - how does it function on novel genomes that are v. distant from refs?


2. How is it useful?
   -

### Maximum-Containment ANI and AAI estimation (tech paper)

- jaccard vs containment vs total shared k-mers
- nucleotide vs protein (vs dayhoff?)

		
### Simulated data

ANI and AAI estimations were assessed on biologically-relevant simulated data from Criscuolo, 2020. 


#### alphabet and k-size selection for optimal distance estimation

### Alignment-free phylogenomic recontruction
_evolpaths analysis_
		




### Maximum-containment AAI enables fast and accurate taxonomic classification

### Comparison with other alignment-free methods (advantages, disadvantages, etc)


Alignment-based metrics are looking at the specific sequence variation of aligned regions, while k-mer based comparisons are comparing shared k-mers vs distinct k-mers. Since each nucleotide polymorphims generates mutated k-mers with an expected frequency, k-mer containment estimates can be used to accurately estimate both the Average Nucleotide Identity and Average Amino Acid Identity  [@doi:10.1101/2021.01.15.426881; @doi:10.1186/s13059-016-0997-x]
